src/leader.py

```diff
@@ -142,14 +148,19 @@ def lead(self):
         Moves the leader vehicle based on controller input.
         """
         code, position = self._controller.get_input()
-        if code == 'right_trigger':
-            self.set_speed(position)
-            self.drive()
-        if code == 'left_trigger':
-            self.set_speed(position)
-            self.reverse()
-        if code == 'dpad_left_right' or code == 'left_stick_x':
-            self.turn(code, position)
+        self._mode = code
+
+        if self._mode in FUNCTION_MAP:
```

Comment:
```python
if code in FUNCTION_MAP
```

This gets rid of the extra variable

Reply:
but also, for this to work, don't we need to keep pressing the same button? It should take in the last pressed button and continue to call that until a different input is received.